src/components/precision-time-picker.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  HelperText,
+  HelperTextItem,
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  Popper,
+  TextInput,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import React from 'react';
+import { padLeadingZero } from '../value-utils';
+
+interface PrecisionTimePickerProps {
+  time: string;
+  onChange?: (time: string, hours?: number, minutes?: number, seconds?: number) => void;
+}
+
+const validTimeStringRegex = /^\d{2}:\d{2}(:\d{2}(\.\d+)?)?$/;
+
+const parseTimeString = (
+  time: string,
+): { hours: number; minutes: number; seconds: number } | null => {
+  if (!validTimeStringRegex.test(time)) {
+    return null;
+  }
+
+  const parts = time.split(':');
+
+  if (parts.length >= 2) {
+    let hours = parseInt(parts[0], 10);
+    hours = hours >= 24 ? 0 : hours;
+    const minutes = parseInt(parts[1], 10);
+    const seconds = parts.length >= 3 ? parseFloat(parts[2]) : 0;
+
+    if (!isNaN(hours) && !isNaN(minutes) && !isNaN(seconds)) {
+      return { hours, minutes, seconds };
+    }
+  }
+
+  return null;
+};
+
+const timeMenuItemsBuilder: () => Array<string> = () => {
+  const items = [];
+
+  for (let hours = 0; hours < 24; hours++) {
+    for (let minutes = 0; minutes < 60; minutes += 30) {
+      items.push(`${padLeadingZero(hours)}:${padLeadingZero(minutes)}`);
+    }
+  }
+
+  return items;
+};
+
+const timeMenuItems = timeMenuItemsBuilder();
+
+export const PrecisionTimePicker: React.FC<PrecisionTimePickerProps> = ({ onChange, time }) => {
+  const [value, setValue] = React.useState(time);
+  const [isValid, setIsvalid] = React.useState(true);
+  const [isMenuOpen, setIsMenuOpen] = React.useState(false);
```

Comment:
Let's use the `useBoolean` hook here. It will remove the need for `openMenu` and `closeMenu`.

Reply:
This fix needs to be compatible with 4.11, so I'm using the SDK version 0.0.12 as seen in [this](https://github.com/openshift/console/tree/master/frontend/packages/console-dynamic-plugin-sdk#openshift-console-versions-vs-sdk-versions) compatibility table.